src/katgpucbf/xbgpu/engine.py

```diff
@@ -583,12 +583,13 @@ async def sender_loop(self) -> None:  # noqa: D102
     def capture_enable(self, *, stream_id: int, enable: bool = True, timestamp: int = 0) -> None:  # noqa: D102
         self.send_stream.enable_beam(beam_id=stream_id, enable=enable, timestamp=timestamp)
 
-    def _weights_updated(self) -> None:
+    def _weights_updated(self) -> int:
```

Comment:
Update the docstring to explain what value is returned. Similarly for the other functions that are now returning a timestamp.